File: web/src/services/api/leaderboard.ts
import { apiGet, compactApiParams } from "@/services/api/request";

export type LeaderboardItem = {
    userId: string;
    username: string;
    displayName: string;
    avatarUrl: string;
    count: number;
};

export type LeaderboardResponse = {
    items: LeaderboardItem[];
};

export async function fetchImageLeaderboard(limit = 50) {
    return apiGet<LeaderboardResponse>("/api/leaderboard/images", compactApiParams({ limit }));
}

File: model/leaderboard.go
package model

// LeaderboardItem 排行榜条目。
type LeaderboardItem struct {
	UserID      string `json:"userId"`
	Username    string `json:"username"`
	DisplayName string `json:"displayName"`
	AvatarURL   string `json:"avatarUrl"`
	Count       int    `json:"count"`
}

type LeaderboardList struct {
	Items []LeaderboardItem `json:"items"`
}
